metrics/value.go

```diff
@@ -39,12 +56,29 @@ func (v *counterVal) Type() string {
 	return "counter"
 }
 
-func (v *counterVal) Values() map[string]interface{} {
-	return map[string]interface{}{
-		"count": v.Count(),
+func (v *counterVal) Keys() iter.Seq[string] {
+	return func(yield func(string) bool) {
+		for _, key := range []string{"count"} {
+			if !yield(key) {
+				return
+			}
+		}
+	}
```

Comment:
Seems like we could decomp out a function that takes the slice of keys and then returns the iterator function.

If each argument slice is a local variable (`counterValKeys = []string{"count"}`, etc.), then calling the helper function should just be a reference to the slice (so no extra allocation/copying), and should also avoid allocation on each iterator call, right? If so, seems like this would be strictly better both from performance and reducing duplicated code.

Reply:
I think you're basically describing [`slices.Values`](https://pkg.go.dev/slices#Values). I tested this and it results in an extra allocation (I think because the slice has to go on the heap to be passed into that function)

```

```